File: tests/automation/pages/analytics_page.py
from playwright.sync_api import Page


class AnalyticsPage:
    def __init__(self, page: Page, base_url: str):
        self.page = page
        self.base_url = base_url

    def goto(self):
        self.page.goto(f"{self.base_url}/analytics")
        self.page.wait_for_load_state("networkidle")

    @property
    def page_heading(self):
        return self.page.get_by_role("heading", name="Analytics").first

    @property
    def total_runs_card(self):
        return self.page.locator("text=Total Runs").first

    @property
    def success_rate_card(self):
        return self.page.locator("text=Success Rate").first

    @property
    def avg_duration_card(self):
        return self.page.locator("text=Avg Duration").first

    @property
    def chart_container(self):
        return self.page.locator("canvas, .recharts-wrapper, [data-testid='chart']").first

    @property
    def export_button(self):
        return self.page.get_by_role("button", name="Export").first

File: tests/automation/pages/AnalyticsPage.ts
import type { Page, Locator } from '@playwright/test';
import { expect } from '@playwright/test';
import { BasePage } from './BasePage.js';

export class AnalyticsPage extends BasePage {
  constructor(page: Page, baseURL: string) {
    super(page, baseURL);
  }

  get charts(): Locator {
    return this.page.locator('.recharts-wrapper, [data-testid="chart"]');
  }

  get dateRangePicker(): Locator {
    return this.page.getByRole('button', { name: /date|range|filter/i }).first();
  }

  get exportButton(): Locator {
    return this.page.getByRole('button', { name: /export/i });
  }

  get kpiCards(): Locator {
    return this.page.locator('[data-testid="kpi-card"]');
  }

  async navigateTo(): Promise<void> {
    await this.goto('/analytics');
    await this.waitForLoad();
  }

  async getChartCount(): Promise<number> {
    return this.charts.count();
  }

  async getKpiCardCount(): Promise<number> {
    return this.kpiCards.count();
  }

  async clickExport(): Promise<void> {
    await this.click(this.exportButton, 'Export button');
  }

  async expectAnalyticsLoaded(): Promise<void> {
    await expect(this.page).toHaveTitle(/analytics|kpi/i);
  }

  async expectChartsVisible(): Promise<void> {
    const count = await this.getChartCount();
    expect(count).toBeGreaterThan(0);
  }
}
